File: backend/src/main/java/com/backend/backend/entity/Order.java
package com.backend.backend.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import lombok.Data;

/**
 * 订单实体类
 */
@Data
@TableName("t_order")
public class Order {

    /**
     * 订单ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 订单号
     */
    @TableField("order_no")
    private String orderNo;

    /**
     * 用户ID
     */
    @TableField("user_id")
    private Long userId;

    /**
     * 场次ID
     */
    @TableField("screening_id")
    private Long screeningId;

    /**
     * 影院ID
     */
    @TableField("cinema_id")
    private Long cinemaId;

    /**
     * 电影ID
     */
    @TableField("movie_id")
    private Long movieId;

    /**
     * 订单总金额
     */
    @TableField("total_amount")
    private BigDecimal totalAmount;

    /**
     * 座位数量
     */
    @TableField("seat_count")
    private Integer seatCount;

    /**
     * 订单状态（PENDING_PAYMENT：待支付，PAID：已支付，CANCELLED：已取消，REFUNDED：已退款，COMPLETED：已完成）
     */
    @TableField("status")
    @EnumValue
    private OrderStatus status;

    /**
     * 支付时间
     */
    @TableField("payment_time")
    private LocalDateTime paymentTime;

    /**
     * 取消时间
     */
    @TableField("cancel_time")
    private LocalDateTime cancelTime;

    /**
     * 创建时间
     */
    @TableField("create_time")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField("update_time")
    private LocalDateTime updateTime;

    // --- 非数据库字段 ---
    @TableField(exist = false)
    private List<OrderSeat> seats; // 用于存储关联的座位信息 (查询时填充)
    @TableField(exist = false)
    private Screening screening; // 用于存储关联的场次信息
    @TableField(exist = false)
    private Movie movie; // 用于存储关联的电影信息

    /**
     * 支付宝交易号
     */
    @TableField(exist = false) // 如果数据库中没有此字段，暂时标记为非数据库字段
    private String alipayTradeNo;
    
    /**
     * 前端回调URL (支付完成后跳转的前端页面地址)
     */
    @TableField(exist = false)
    private String frontendCallbackUrl;

    // 定义订单状态枚举
    public enum OrderStatus {
        PENDING_PAYMENT, // 待支付
        PAID,            // 已支付
        CANCELLED,       // 已取消
        REFUNDED,        // 已退款 (预留)
        COMPLETED        // 已完成 (已取票/观影)
    }
} 

File: frontend/src/types/order.ts
/**
 * Represents the identifier for a single seat.
 * Used in CreateOrderRequest.
 * Assumes 1-based indexing as required by the backend API.
 */
export interface SeatInfo {
  rowIndex: number // 1-based
  colIndex: number // 1-based
}

/**
 * Request body for creating a new order.
 */
export interface CreateOrderRequest {
  screeningId: number
  selectedSeats: SeatInfo[] // Array of 1-based seat identifiers
}

/**
 * Represents a single seat within an order.
 */
export interface OrderSeat {
  id: number
  orderId: number
  screeningId: number
  rowIndex: number // Usually 1-based from backend
  colIndex: number // Usually 1-based from backend
  seatLabel?: string // e.g., "5排7座"
}

/**
 * Represents the detailed Order object, often returned by GET /api/orders/{id}.
 */
export interface Order {
  id: number
  orderNo: string
  userId: number
  screeningId: number
  cinemaId: number
  movieId: number
  totalAmount: number // Renamed from totalPrice in example for consistency?
  seatCount: number
  status: 'PENDING_PAYMENT' | 'PAID' | 'CANCELLED' | 'REFUNDED' | 'COMPLETED'
  paymentTime?: string | null // ISO 8601 or null
  cancelTime?: string | null // ISO 8601 or null
  createTime: string // ISO 8601
  updateTime?: string // ISO 8601
  seats?: OrderSeat[] // Use OrderSeat type
  // Optional embedded details (adjust based on actual API response structure)
  movieTitle?: string
  cinemaName?: string
  roomName?: string
  screeningTime?: string // ISO 8601
}

/**
 * Represents the Order View Object, often returned after creating an order (POST /api/orders)
 * or in order lists.
 * Based on the OrderVO example in the API documentation.
 */
export interface OrderVO {
  // Rename orderId to id to match assumed API response and template usage
  id: number
  orderNo: string
  userId: number
  screeningId: number
  movieId: number
  movieTitle: string
  cinemaId: number
  cinemaName: string
  roomName: string
  screeningStartTime: string // ISO 8601
  totalAmount: number
  seatCount: number
  status: 'PENDING_PAYMENT' | 'PAID' | 'CANCELLED' | 'REFUNDED' | 'COMPLETED'
  paymentTime?: string | null // ISO 8601 or null
  cancelTime?: string | null // ISO 8601 or null
  createTime: string // ISO 8601
  seatsDescription?: string[] // e.g., ["5排7座", "5排8座"]
}
